### ISSUE TITLE
Enhancement: [Inline Editable] Add appearance="minimal" property

### ISSUE THREAD

**Issue by capeoples** [Esri/calcite-design-system#1597]
**monday.com sync:** #18048538017

### Description
For the inline editable component, it would be ideal to be able to update the button styles for the "x" and "✔️" to be any supported button style. This will enable this component to fully adapt to whatever design/workflow is needed. 
<img width="395" alt="Screen Shot 2021-02-22 at 1 23 50 PM" src="https://user-images.githubusercontent.com/30473086/108772180-85865800-7511-11eb-9a59-cf8cb63d4b9c.png">

#### Which Component
Inline editable > controls enabled

#### Example Use Case
In the create new item workflow in Online, a user could want to create a new folder to store the item in. We need the confirm/cancel icons to support this, but it would not be appropriate for the "✔️" to use the solid button style as it's contained in a workflow that already has a primary/solid action button. 

cc @paulcpederson

**Comment by macandcheese** [comment]
Is this still a valid request @capeoples ?
If so, would the iconography need to be customizable as well?

**Comment by macandcheese** [comment]
Could your team provide examples of what they’d need visually? 

Choosing from a set of icons (like the split-button experience where you select from a pre existing set - in this case add, check, etc)or more control than that?

A similar question for color of primary button - we could allow a choice for the main button from our set of button colors (besides red).

Short of slotting custom buttons it would be nice to have at least some set of pre defined options to keep these consistent even when teams need to adjust slightly.

**Comment by paulcpederson** [comment]
I think the main case is a secondary interface on the calcite background color with more transparent elements (would be great to have a more minimal input style for this as well). A lot of times it's a list where every single element has an inline edit functionality. (Editing abcabc folder name below)

![Screen Shot 2022-04-22 at 8 25 03 AM](https://user-images.githubusercontent.com/1031758/164745691-8e123e5f-2d76-49b9-91b2-646b0cc11c35.png)

You could argue that we could use the inline edit as is here, but it feels pretty aggressive.

**Comment by macandcheese** [comment]
Makes sense... some thoughts / options:

In the past we've heard concerns over a "minimal" input type - I agree that would be a nice solution to leverage here. Maybe that is something we can re-visit, since it applies in this case but also in the "normal" input component.

Would you be ok deferring to the provided icons at all times (read: does the check / clear ever need to be customizable)?

Would you be ok if there was a prescribed "secondary / subtle" inline-editable appearance that allowed another visual option but stopped short of explicit customizability of each button?

**Comment by paulcpederson** [comment]
> Would you be ok deferring to the provided icons at all times

Yeah, that's fine. Would be better to be consistent about the icons used anyways. 

> Would you be ok if there was a prescribed "secondary / subtle" inline-editable appearance that allowed another visual option but stopped short of explicit customizability of each button?

Yes. I think it's probably easier if calcite just provides a slightly more subtle style. Personally I don't need super fine grained control over each button, just a way to turn the volume knob down a bit in general.

**Comment by macandcheese** [comment]
Excellent. Let's repurpose this issue as "add a secondary / subtle" appearance style to the inline-editable component. This would align with other components that have an "appearance" prop, where we provide consistent alternatives to our default styles.

**Comment by paulcpederson** [comment]
Nice, in my mind this is very similar to something like the accordion's `appearance="minimal"` prop. This is a property that would be welcome on lots of components like lists, etc.

**Comment by macandcheese** [comment]
I think we should close this, and if desired once we built in editable functionality to individual components (https://github.com/Esri/calcite-design-system/issues/6153), open a new, updated issue. cc @Esri/calcite-designers
